### ISSUE TITLE
Add support for presets in minimum height fields

### ISSUE THREAD

**Issue by cbirdsong** [WordPress/gutenberg#48810]
The cover block has long supported entering custom minimum height values, and support was just added to the group and post content blocks: https://make.wordpress.org/core/2023/03/06/minimum-height-dimensions-block-support/

However, entering in a simple custom value rarely works well for responsive web design. `vh` units can work decently, but the field defaults to `px` and the particulars of the different CSS unit options are not easy for content editors to understand. On top of that, you really need to be able to use `min`, `max` and `clamp` values for a great responsive design. 

To address that, the editor should support registering custom presets for the height field. The default editing experience should have the user pick from a list of presets with an option to enter a custom value, and themes should be able to disable entry of custom values. This would make the height field work exactly like spacing fields. 

This could also replace the "Toggle full height" button that switches the cover block to full height since "full" could be one of the presets.

**Comment by jasmussen** [comment]
Catching up on this one, better late than never. I'm wondering: is there an angle on this where we focus on the "presets" aspect, and make the "minimum height" aspect secondary? What I'm hearing is a desire for custom presets that might include things like min/max/clamp. But would this not be valid also for other fields? E.g. beyond the min-height of a Cover, perhaps the Spacer block, or even the margins on a Separator block or anything else?

Asking because there might be a way to reframe (perhaps re-title) this issue, and any designs we might ponder (cc: @WordPress/gutenberg-design) could be contextualised better by these presets being available for _any_ field. 

Consider the unitcontrol—click "px" and you can choose another unit. Could we simply add a "Custom value" option at the bottom which opened a modal that let you type in the full clamp value manually?

**Comment by cbirdsong** [comment]
The ability to type in a full clamp value would be nice as an option, but I don't think it would obviate the need for presets.

What you get with presets:
- the ability to offer a limited and controlled number of options so your editors can't wildly deviate from the approved design system
- the ability to generate your own CSS classes instead of inline CSS if you need to have a strict content security policy

Additionally, I imagine you would want a drastically different set of options for width/height presets compared to padding/margin, though there might be some overlap in some cases.

**Comment by jasmussen** [comment]
Good point. What about an option in the dropdown called "Apply preset" which would also open a modal, but this modal would include the presets you could choose from?

**Comment by t-hamano** [comment]
@cbirdsong Have you tried the `dimensionSizes` property introduced in WordPress 7.0 yet? This setting is essentially a preset for width and height.

https://make.wordpress.org/core/2026/03/15/dimensions-support-enhancements-in-wordpress-7-0/

The presets defined there are automatically selectable in the following blocks:

- `core/button` — `width`
- `core/icon` — `width`
- `core/group` — `minHeight`, `minWidth`
- `core/post-content` — `minHeight`
- `core/pullquote` — `minHeight`
- `core/quote` — `minHeight`
- `core/verse` — `minHeight`

However, please note that while the Cover block supports minimum height, this preset is not available because the Cover block's minimum height is a legacy, proprietary block attribute.

**Comment by t-hamano** [comment]
I am trying to migrate the minHeight attribute of the Cover block to block supports so that it can also benefit from presets. https://github.com/WordPress/gutenberg/pull/83447

**Comment by cbirdsong** [comment]
> Have you tried the `dimensionSizes` property introduced in WordPress 7.0 yet? This setting is essentially a preset for width and height.

Not yet, but I'm happy to see movement in this direction!

> Good point. What about an option in the dropdown called "Apply preset" which would also open a modal, but this modal would include the presets you could choose from?

I would expect it to pretty much work like the existing presets. The default/obvious experience should be selecting from presets, entry of custom values is nested behind a control of some kind, which [can also be disabled with a key in theme.json](https://developer.wordpress.org/themes/global-settings-and-styles/settings/color/#user-customized-colors). 

Size values in particular _greatly_ benefit from presets being the default happy path because they can contain `clamp`-based responsive sizes, and users editing on desktop frequently do not think about what a minimum height of 800px will look like on a mobile device. 

The spacer block currently stumbles pretty hard here, defaulting to a custom value and only showing presets if you toggle over to it:

https://github.com/user-attachments/assets/a13e8ec2-72ca-4d62-bb41-ed40f76ed3e7

The drag handle also very easily discards a preset and switches back to a custom value:

https://github.com/user-attachments/assets/9c6f6cc0-1f58-4a4b-809e-4c32683ec64e

There's more discussion about this in https://github.com/WordPress/gutenberg/issues/54084.
